# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: image/VectorImage.cpp
# Commit: 57d7a634ef8f
# Full Hash: 57d7a634ef8ff28946c3cb92b261f32308b344a0
# Author: Edgar Chen <echen@mozilla.com>
# Date: 2019-08-22 04:04:03
# Regressor Bug: 1568950
# File Overlap Count: 1
# Description:
#   Bug 1568950 - Part 2: Record use counters for SVG image when the image is going to be discarded; r=smaug
#   
#   Part-1 patch (moving ReportUseCounters call to Document::Destroy) doesn't help SVG image case because the SVGDocumentWrapper::DestroyViewer() still happens too late.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D42368
# ==============================================================================

diff -r 81f72abb79ff -r 57d7a634ef8f image/VectorImage.cpp
--- a/image/VectorImage.cpp	Wed Aug 21 12:59:56 2019 +0000
+++ b/image/VectorImage.cpp	Wed Aug 21 12:58:12 2019 +0000
@@ -1283,6 +1283,10 @@
     mProgressTracker->OnDiscard();
   }
 
+  if (Document* doc = mSVGDocumentWrapper->GetDocument()) {
+    doc->ReportUseCounters();
+  }
+
   return NS_OK;
 }
 
